x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_list/components/alerts_list.tsx

```diff
@@ -136,30 +136,33 @@ export const AlertsList: React.FunctionComponent = () => {
   }, []);
 
   async function loadAlertsData() {
-    setAlertsState({ ...alertsState, isLoading: true });
-    try {
-      const alertsResponse = await loadAlerts({
-        http,
-        page,
-        searchText,
-        typesFilter,
-        actionTypesFilter,
-      });
-      setAlertsState({
-        isLoading: false,
-        data: alertsResponse.data,
-        totalItemCount: alertsResponse.total,
-      });
-    } catch (e) {
-      toastNotifications.addDanger({
-        title: i18n.translate(
-          'xpack.triggersActionsUI.sections.alertsList.unableToLoadAlertsMessage',
-          {
-            defaultMessage: 'Unable to load alerts',
-          }
-        ),
-      });
-      setAlertsState({ ...alertsState, isLoading: false });
+    const hasAnyAuthorizedAlertType = alertTypesState.data.size > 0;
+    if (hasAnyAuthorizedAlertType) {
```

Comment:
curious there's no `else` clause for this `if`.  This function appears to just update state after loading the alerts, and that doesn't need to be done if there's nothing to view.  But then wondering if the user will end up at with an empty view, or perhaps even _gulp_ a view which suggests to create a new alert :-).  Guessing this case is covered already, just not clear from looking **just** at this function, will test it live.

Reply:
Correct, it's handled in the render itself - which is why I would like us to split this file down into like 4 or 5 files - it's way too big and hard to follow.

Reply:
But that can be a separate PR